xwiki-platform-core/xwiki-platform-eventstream/xwiki-platform-eventstream-api/src/main/java/org/xwiki/eventstream/events/AbstractEventStreamEvent.java

```diff
@@ -27,6 +27,12 @@
  */
 public abstract class AbstractEventStreamEvent implements org.xwiki.observation.event.Event
 {
+    /**
+     * Used to provide a key to a property in the current execution context that avoids stepping into a loop when
+     * triggering new events.
+     */
+    public static final String EVENT_LOOP_CONTEXT_LOCK_PROPERTY = "eventLoopContextLockProperty";
```

Comment:
I’m not sure that it’s a good idea to put the property in an internal package : this could be accessed by an external listener for any kind of computation.
Also, it would mean that we’re using internal properties of the Event Stream from the Activity Stream module.